src/coreclr/jit/emit.h

```diff
@@ -1381,13 +1391,18 @@ class emitter
 #if FEATURE_LOOP_ALIGN
     struct instrDescAlign : instrDesc
     {
-        instrDescAlign* idaNext; // next align in the group/method
-        insGroup*       idaIG;   // containing group
-    };
-
-    void emitLoopAlign(unsigned short paddingBytes);
-    void emitLongLoopAlign(unsigned short alignmentBoundary);
+        instrDescAlign* idaNext;     // next align in the group/method
+        insGroup*       idaIG;       // containing group
+        insGroup*       idaTargetIG; // The IG before the loop IG.
+                                     // If no 'jmp' instructions were found until targetIG,
+                                     // the 'align' will be placed in this IG.
```

Comment:
> I'm confused by this description. For one, this is an instrDesc of an align instruct that is already in an IG, so "will be placed in this IG" doesn't make sense.

I think better way to put is: "If no 'jmp' instructions were found until targetIG, then `idaTargetIG` will be same as `idaIG`." Let me know if that makes sense.


> Does the align instruction get placed in a JMP IG, with idaTargetIG pointing the the IG before the loop -- the "last possible place the alignment can be placed", and then we might later generate an align instruction into a subsequent JMP IG that still precedes the target?

This logic has been changed now because of separate "loop alignment placement" phase. The IG, either the one that ends with `jmp` or the one before the loop, will be determine in before code generation and the `align` will be placed in the corresponding IG. 

For scenarios where `align` is placed in a different IG (`IG03`) that doesn't precede the loop that starts at `IG06`, `idaTargetIG == IG05` and `idaIG == IG03`.

```asm
IG03:
      ...
      jmp IG19
                ; <---- align instruction placement
IG04:
    ...

IG05
   ...
   jne IG08

IG06:
  ...
  ...

IG07:
  ...
   jne IG06
  
```


For scenarios where `align` is placed in a immediately before the IG with loop that starts at `IG05`, `idaTargetIG == idaIG == IG04`.


```asm
IG04:
    ...
    jne IG08
                ; <---- align instruction placement
IG05
   ...
   ...

IG06:
  ...
  jne IG05

```

`idaTargetIG` will be point to the if `align` was placed in non-preceding IG

> The igTargetIG points forwards in the code stream, right? So is it null for a while, then filled in?

Correct. It is filled in `emitConnectAlignInstrWithCurIG()`.